cpp/src/arrow/compute/api_scalar.h

```diff
@@ -77,6 +77,18 @@ struct ARROW_EXPORT SplitPatternOptions : public SplitOptions {
   std::string pattern;
 };
 
+struct ARROW_EXPORT ReplaceSliceOptions : public FunctionOptions {
+  explicit ReplaceSliceOptions(int64_t start, int64_t stop, std::string replacement)
+      : start(start), stop(stop), replacement(std::move(replacement)) {}
+
+  /// Index to start slicing at
+  int64_t start = 0;
+  /// Index to stop slicing at
+  int64_t stop = std::numeric_limits<int64_t>::max();
```

Comment:
Hmm... I'm not sure the default values will be picked up. Is it just for documentation?